parsl/executors/taskvine/executor.py

```diff
@@ -232,145 +115,171 @@ class TaskVineExecutor(BlockProviderExecutor, putils.RepresentationMixin):
     @typeguard.typechecked
     def __init__(self,
                  label: str = "TaskVineExecutor",
-                 project_name: Optional[str] = None,
-                 project_password_file: Optional[str] = None,
-                 address: Optional[str] = None,
-                 port: int = VINE_DEFAULT_PORT,
-                 env: Optional[Dict] = None,
-                 shared_fs: bool = False,
-                 use_cache: bool = True,
-                 source: bool = False,
-                 pack: bool = False,
-                 pack_env: str = "",
-                 extra_pkgs: Optional[List[str]] = None,
-                 autolabel: bool = False,
-                 autolabel_window: int = 1,
-                 autocategory: bool = True,
-                 max_retries: int = 1,
-                 init_command: str = "",
-                 worker_options: str = "",
-                 worker_executable: str = 'vine_worker',
-                 function_dir: Optional[str] = None,
-                 wait_for_workers: int = 0,
-                 enable_peer_transfers: bool = True,
-                 full_debug: bool = False,
-                 provider: ExecutionProvider = LocalProvider(),
+                 use_factory: bool = False,
+                 manager_config: TaskVineManagerConfig = TaskVineManagerConfig(),
+                 factory_config: TaskVineFactoryConfig = TaskVineFactoryConfig(),
+                 provider: ExecutionProvider = LocalProvider(init_blocks=1),
                  storage_access: Optional[List[Staging]] = None):
+
+        # If TaskVine factory is used, disable the Parsl provider
+        if use_factory:
+            provider = LocalProvider(init_blocks=0, max_blocks=0)
```

Comment:
have you tried using `None` here? ideally that would work here but maybe it doesn't...

Reply:
Using `None` almost works, until the intepreter exits...
```
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/jason/work/parsl_taskvine/parsl/parsl/process_loggers.py", line 27, in wrapped
    r = func(*args, **kwargs)
  File "/home/jason/work/parsl_taskvine/parsl/parsl/dataflow/dflow.py", line 1205, in cleanup
    job_ids = executor.provider.resources.keys()
AttributeError: 'NoneType' object has no attribute 'resources'
```

An easy fix is to check for existence of `executor.provider` and is coporated in this PR. `provider` is set to `None`.

Reply:
mypy doesn't allow assignment to None so revert back to dummy LocalProvider.

Reply:
ok - I'll see if I can sort that out in a different PR...

Reply:
I made PR https://github.com/Parsl/parsl/pull/2825 to work on allowing this